x-pack/plugins/security_solution/server/lib/detection_engine/routes/__mocks__/server.ts

```diff
@@ -12,28 +12,48 @@ import { responseMock as responseFactoryMock } from './response_factory';
 import { requestMock } from '.';
 import { responseAdapter } from './test_adapters';
 import type { SecuritySolutionRequestHandlerContext } from '../../../../types';
+import type { RegisteredVersionedRoute } from '@kbn/core-http-router-server-mocks';
 
 interface Route {
-  config: RouteConfig<unknown, unknown, unknown, 'get' | 'post' | 'delete' | 'patch' | 'put'>;
+  validate: RouteConfig<
+    unknown,
+    unknown,
+    unknown,
+    'get' | 'post' | 'delete' | 'patch' | 'put'
+  >['validate'];
   handler: RequestHandler;
 }
 
-const getRoute = (routerMock: MockServer['router']): Route => {
-  const routeCalls = [
-    ...routerMock.get.mock.calls,
-    ...routerMock.post.mock.calls,
-    ...routerMock.put.mock.calls,
-    ...routerMock.patch.mock.calls,
-    ...routerMock.delete.mock.calls,
-  ];
-
-  const [route] = routeCalls;
-  if (!route) {
-    throw new Error('No route registered!');
+const HTTP_METHODS = ['get', 'post', 'put', 'patch', 'delete'] as const;
+
+const getClassicRoute = (routerMock: MockServer['router']): Route | undefined => {
+  const method = HTTP_METHODS.find((m) => routerMock[m].mock.calls.length > 0);
+  if (!method) {
+    return undefined;
   }
 
-  const [config, handler] = route;
-  return { config, handler };
+  const [config, handler] = routerMock[method].mock.calls[0];
+  return { validate: config.validate, handler };
+};
+
+const getVersionedRoute = (router: MockServer['router']): Route => {
+  const method = HTTP_METHODS.find((m) => router.versioned[m].mock.calls.length > 0);
+  if (!method) {
+    throw new Error('No route registered!');
+  }
+  const config = router.versioned[method].mock.calls[0][0];
+  const routePath = config.path;
+
+  const route: RegisteredVersionedRoute = router.versioned.getRoute(method, routePath);
+  const firstVersion = Object.values(route.versions)[0];
+
+  return {
+    validate:
+      firstVersion.config.validate === false
```

Comment:
can we do 
`validate: firstVersion.config.validate?.request || false`  instead?

Reply:
`firstVersion.config.validate` is typed as `false | { request: ... }`. Because of this, we have to explicitly check if the value is `false` before accessing any object properties.